### ISSUE TITLE
BlueStacks 5.20 Root Instructions

### ISSUE THREAD

**Issue by VisionR1** [HuskyDG/magisk-files#97]
I just trim the instructions and test it

I don't know if this method is good because chance from ReadOnly to Normal but because BSTweaker 6.9.2 beta not work with Latest 5.20, a quick method and old methot from 2021 from xda here: https://xdaforums.com/t/guide-magisk-supersu-and-xposed-for-memu-7-3-2-or-ldplayer-4-0-45.4222919/post-85009977  (i test it and still work)

1) Need to make /system partition writable first.
For that, you need to go to this directory "%ProgramData%\BlueStacks_nxt\Engine\Rvc64"
This changing for the master instance Rvc64.
Edit the file Rvc64.bstk and change type="ReadOnly" to type="Normal" for Root.vdi

2) Need Su Access.
For this go to "%ProgramData%\BlueStacks_nxt" and edit the file bluestacks.conf present within.
Search for root and look for an entry similar to - bst.instance.Rvc64.enable_root_access="0"
That entry is for the master instance of Rvc64. Change it's value to 1 and save this file

Note: Make sure you made these changes while there were no instances running and no services related to bluestacks were running, if there are please close or kill them via task manager

If all good start the emulator and try install latest Kitsune to see if it work

![BlueStacks 5 20 Root](https://github.com/HuskyDG/magisk-files/assets/25982450/1c4d179e-0d54-47e6-851b-1aeaf8d5e6d4)

**Comment by Jordan231111** [comment]
Can confirm it works, but files may be different for everyone make to sure edit the 3 below to Normal 
       with quotes.
       
       <HardDisk uuid="{redacted}" location="fastboot.vdi" format="VDI" type="Normal"/>
        <HardDisk uuid="{redacted}" location="Root.vhd" format="VHD" type="Normal"/>
        <HardDisk uuid="{redacted}" location="Data.vhdx" format="VHDX" type="Normal"/>

**Comment by VisionR1** [comment]
> ```
>    Can confirm it works, but files may be different for everyone make to sure edit the 3 below to Normal 
>    with quotes.
>    
>    <HardDisk uuid="{redacted}" location="fastboot.vdi" format="VDI" type="Normal"/>
>     <HardDisk uuid="{redacted}" location="Root.vhd" format="VHD" type="Normal"/>
>     <HardDisk uuid="{redacted}" location="Data.vhdx" format="VHDX" type="Normal"/>
> ```

Besides that, the Data.vhdx is normal by default, and the fastboot.vdi why normal ?

**Comment by Jordan231111** [comment]
> > ```
> >    Can confirm it works, but files may be different for everyone make to sure edit the 3 below to Normal 
> >    with quotes.
> >    
> >    <HardDisk uuid="{redacted}" location="fastboot.vdi" format="VDI" type="Normal"/>
> >     <HardDisk uuid="{redacted}" location="Root.vhd" format="VHD" type="Normal"/>
> >     <HardDisk uuid="{redacted}" location="Data.vhdx" format="VHDX" type="Normal"/>
> > ```
> 
> Besides that, the Data.vhdx is normal by default, and the fastboot.vdi why normal ? I ask because even with BlueStacksTweaker still remain Readonly

that doesn't really harm it to keep it at normal also easier to make one click root solution like bstweaker
that should work for all android 11 bs 5 versions unless they change the file format drastically

https://github.com/Jordan231111/BluestacksRoot/releases/tag/Bluestacks_5.20_Latest

**Comment by VisionR1** [comment]
> > > ```
> > >    Can confirm it works, but files may be different for everyone make to sure edit the 3 below to Normal 
> > >    with quotes.
> > >    
> > >    <HardDisk uuid="{redacted}" location="fastboot.vdi" format="VDI" type="Normal"/>
> > >     <HardDisk uuid="{redacted}" location="Root.vhd" format="VHD" type="Normal"/>
> > >     <HardDisk uuid="{redacted}" location="Data.vhdx" format="VHDX" type="Normal"/>
> > > ```
> > 
> > 
> > Besides that, the Data.vhdx is normal by default, and the fastboot.vdi why normal ? I ask because even with BlueStacksTweaker still remain Readonly
> 
> that doesn't really harm it to keep it at normal also easier to make one click root solution like bstweaker that should work for all android 11 bs 5 versions unless they change the file format drastically
> 
> https://github.com/Jordan231111/BluestacksRoot/releases/tag/Bluestacks_5.20_Latest

Yeah i get your point, if not harm it all good, thanks for create this .bat file, one click root solution is more easy for beginner, and for the rest of us the old method still teach us. Old is gold

**Comment by david8557** [comment]
I'm using this root method on BlueStacks 5.21.0 but BS5 always crash,not responding after a while. Does anyone have that problem?

**Comment by Jordan231111** [comment]
> I'm using this root method on BlueStacks 5.21.0 but BS5 always crash,not responding after a while. Does anyone have that problem?

Likely because u manually edited it wrong, thus use the script to prevent mistakes. Likely have to undo all ur edits
or reinstall the program
